### PR TITLE
Update branch PR bot to only push --force when creating a new PR

### PR DESCRIPTION
**Pull request by natemcmaster**
This will no longer overwrite changes maintainers may push to the bot's PR to resolve conflicts.

cc @weshaggard

### CODE REVIEW

File: scripts/GitHubMergeBranches.ps1 [hunk 1]
```diff
@@ -272,8 +268,29 @@ try {
...
+        }
+        catch {
+            Write-Warning "Failed to update existing PR"
+        }
+
+        $prMessage = if ($prUpdatedSuccess) {
+            "This pull request has been updated.`n`n$committersList"
+        } else {
+            @"
+:x: Uh oh, this pull request could not be updated automatically. New commits were pushed to $HeadBranch, but I could not automatically push those to $mergeBranchName to update this PR.
```
Comment (anurse) [inline scripts/GitHubMergeBranches.ps1]: Clearly our bot should speak in GIFs :)

File: scripts/GitHubMergeBranches.ps1 [hunk 2]
```diff
@@ -272,8 +268,29 @@ try {
...
+You may need to fix this problem by merging branches with this PR. Contact the ASP.NET Core Engineering if you are not sure what to do about this.
```
Comment (anurse) [inline scripts/GitHubMergeBranches.ps1]: nit:

Contact the ASP.NET Core Engineering **Team**...

or

Contact ~~the~~ ASP.NET Core Engineering ...

File: scripts/GitHubMergeBranches.ps1 (hunk 1 above)
Comment (natemcmaster) [reply scripts/GitHubMergeBranches.ps1]: Haha

File: scripts/GitHubMergeBranches.ps1 (hunk 2 above)
Comment (natemcmaster) [reply scripts/GitHubMergeBranches.ps1]: good catch. updated.

**Pull request merged**